### PR TITLE
Feat:package.json update

### PR DESCRIPTION
**Pull request by Trent1900**
### Descriptions

- Link to UI design (if possible):
- https://github.com/taodemy/website/issues/69

### Provide screenshots or videos that the function is working.
image

### Provide screenshots of no warning, or error in the console.
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

**Comment by Trent1900** [comment]
It seems very difficult to use environment variables in package.json or other json files, that is the reason why I did not change the port 3001 into environment variables. 
Please show me how to do that.
I have did some researches on stackoverflow:

I also try asking chatgtp, and failed with his first suggestion, and  then I asked if he is sure about his answer, while he seems not sure about that(sad). lol
image

so please shed some light on the question, to prove that human can do better job than chatgpt, thank you.

**Comment by Trent1900** [comment]
I am not sure if it is good idea to bind the json server script onto build command, or should I make it start command. 
ps: I did it because on Vercel, there is always a build when CI/CD pipe line deploy, so that the server will start when build.

**Comment by Trent1900** [comment]
with the Vercel build command changed, we are able to see the server is running from vercel log.
image

Still, I am not sure if this is the best way or right way to do it, let me know if there is a another way. 
thanks.

**Comment by tatoMa** [comment]
> 

Surely, it is not the best way at all. Ultimately, we need a backend to handle all the APIs but I want us to focus on the Front-end mainly, at least during this 3-month long session.

**Comment by tatoMa** [comment]
> It seems very difficult to use environment variables in package.json or other json files, that is the reason why I did not change the port 3001 into environment variables. Please show me how to do that. I have did some researches on stackoverflow:
> 
> I also try asking chatgtp, and failed with his first suggestion, and then I asked if he is sure about his answer, while he seems not sure about that(sad). lol image
> 
> so please shed some light on the question, to prove that human can do better job than chatgpt, thank you.

What about defining an env variable and using this variable on one of scripts on package.json?

Give it a try. If it is not achievable, maybe adding a new repo with only files dealing with json-server, and deploying it in a separate Vercel project can be a better idea.

**Comment by Trent1900** [comment]
> 



> > It seems very difficult to use environment variables in package.json or other json files, that is the reason why I did not change the port 3001 into environment variables. Please show me how to do that. I have did some researches on stackoverflow:
> > I also try asking chatgtp, and failed with his first suggestion, and then I asked if he is sure about his answer, while he seems not sure about that(sad). lol image
> > so please shed some light on the question, to prove that human can do better job than chatgpt, thank you.
> 
> What about defining an env variable and using this variable on one of scripts on package.json?
> 
> Give it a try. If it is not achievable, maybe adding a new repo with only files dealing with json-server, and deploying it in a separate Vercel project can be a better idea.

according to stackover flow [link](https://stackoverflow.com/questions/25112510/how-to-set-environment-variables-from-within-package-json), I have managed to set the environment variable in package.json. while, the envrionment file is hard coded according to the link, so we will need two scripts in package.json for local and production respective, i.e to use .env and .env.local respectively. 
let me know if this works for our project or not.

**Comment by Trent1900** [comment]
With the help of `dotenv`, the`PORT` variable has been set in the environment. Additionally, a `.env.example` file has been created. For local development, please rename `env.example` to `.env` and modify the port to the desired one on your local machine. For instance, if you want the JSON server to run on port 3003, change the .env file to be `JSON_SERVER_PORT=3001`

changes also be made: 

1. Modified build and dev commands to run server first.
2. Used environment variables on Vercel (see picture).

3821677501312_ pic

**Review by HarryuCATGREY — APPROVED** [review]
LGTM

File: .env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+JSON_SERVER_PORT={"put your desirable port here"}
```
Comment (tatoMa) [inline .env.example:1]: Nice example.

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request closed without merging**